File: src/types/search-filters.ts
/**
 * Search filters for conversations and snippets
 */
export interface SearchFilters {
  source?: 'chatgpt' | 'claude' | 'perplexity' | 'kimi' | 'other';
  tags?: string[];
  collection_id?: number;
  type?: 'conversation' | 'snippet' | 'all';
}

/**
 * Snippet filters
 */
export interface SnippetFilters {
  language?: string;
  tags?: string[];
  source_conversation_id?: number;
}

File: server/application/internal/models/search_filters.go
package models

// SearchFilters represents filters for searching conversations
type SearchFilters struct {
	Query        string
	Source       string
	Tags         []string
	CollectionID *int
}

// SnippetFilters represents filters for listing snippets
type SnippetFilters struct {
	Language             string
	Tags                 []string
	SourceConversationID *int
}

